### ISSUE TITLE
selinux context of symlinks not set

### ISSUE THREAD

**Issue by laolux** [containers/podman#14145]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

podman does not correctly set the selinux label of symlinks.

**Steps to reproduce the issue:**

1. create a directory with a file and symlink, e.g. ~/data/file1 and ~/data/symlink1 pointing to file1

2. run a container `podman run --rm -it -v ~/data:/data:Z docker.io/library/alpine:latest /bin/sh`

3. In the container, try to read the symlink `cat /data/symlink1`, it will fail with `Permission denied`

4. As root, check for AVC messages: `ausearch -m AVC,USER_AVC -ts recent`. There should be something like
`type=AVC msg=audit(1651931724.169:525): avc:  denied  { read } for  pid=11068 comm="cat" name="symlink1" dev="dm-1" ino=14627025 scontext=system_u:system_r:container_t:s0:c387,c766 tcontext=unconfined_u:object_r:user_home_t:s0 tclass=lnk_file permissive=0

5. On the host, check the selinux labels for `~/data/file1` and `~/data/symlink1`. file1 should have something like `system_u:object_r:container_file_t:s0:c282,c443`, wheres symlink1 will be unchanged from creation and look like `unconfined_u:object_r:user_home_t:s0`

**Describe the results you received:**
podman does not set the selinux context of the symlink.

**Describe the results you expected:**
podman correctly sets the selinux context of the symlink.

**Additional information you deem important (e.g. issue happens only occasionally):**
podman version 3.4.7 used to set the selinux context of symlinks correctly.

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.0.3
API Version:  4.0.3
Go Version:   go1.18
Built:        Sat Apr  2 03:21:54 2022
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.3
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 4
  distribution:
    distribution: fedora
    version: "36"
  eventLogger: journald
  hostname: server
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.17.5-300.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 129495040
  memTotal: 8159903744
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.4-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.4
      commit: 6521fcc5806f20f6187eb933f9f45130c86da230
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8138256384
  swapTotal: 8158965760
  uptime: 1h 10m 33.17s (Approximately 0.04 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/laolux/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/laolux/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  volumePath: /home/laolux/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.3
  Built: 1648837314
  BuiltTime: Sat Apr  2 03:21:54 2022
  GitCommit: ""
  GoVersion: go1.18
  OsArch: linux/amd64
  Version: 4.0.3
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.0.3-1.fc36.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

No, but the release notes of 4.1.0 do not mention any work related to selinux. I will check with 4.1.0 soon when it is easier to install (new dependencies `netavark` and who knows what that will entail).

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by laolux** [comment]
Fixed in 4.1.0, though absent from release notes.

**Comment by SvenVD** [comment]
I also experienced this bug, any workaround for fixing this manually, until podman 4.1.0 lands in RHEL8?
